.github/workflows/workflow.yml

```diff
@@ -13,42 +13,123 @@ on:
       - '**.md'
 
 jobs:
-  test:
-    name: Test on ${{ matrix.os }}
-
+  # Build and unit test
+  build:
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest, macOS-latest]
       fail-fast: false
-
     runs-on: ${{ matrix.os }}
-
     steps:
-    - uses: actions/checkout@v1
-
-    - uses: actions/setup-node@v1
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Setup Node.js
+      uses: actions/setup-node@v1
       with:
         node-version: '12.x'
-
-    - name: Get npm cache directory
+    - name: Determine npm cache directory
       id: npm-cache
       run: |
         echo "::set-output name=dir::$(npm config get cache)"
-
-    - uses: actions/cache@v1
+    - name: Restore npm cache
+      uses: actions/cache@v1
       with:
         path: ${{ steps.npm-cache.outputs.dir }}
         key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
         restore-keys: |
           ${{ runner.os }}-node-
-
```

Comment:
i removed the line breaks between steps since a lot more content is in the yaml file now (tighten up job; make file more skimmable)

Reply:
hello